ansible/roles/products/common/hand_config/tasks/main.yml

```diff
@@ -40,14 +35,30 @@
           file_changed=false
         fi
 
+        git_pull_with_timeout() {
+          # Colours for echo
+          ORANGE='\033[0;33m'
+          GREEN='\033[0;32m'
+          NC='\033[0m' # No Color
+          timeout 5s git pull
+          retval=$?
+          if [ $retval -eq 124 ]; then
+            echo -e "${ORANGE}WARNING: git pull timed out after five seconds, NOT pulling latest hand calibration${NC}"
+          elif [ $retval -eq 1 ]; then
+            echo -e "${GREEN}git pull did not timeout${NC}"
```

Comment:
Maybe it would be clearer if we say:
"git pull finished successfully"

Reply:
Otherwise it may sound like it was expected to time out and it didn't.

Reply:
I did have that originally, but it turns out that `git pull` can fail in a few different ways but still return `1`

Reply:
oh no, my bad

Reply:
nice, thanks
c3c10915a570e56bb31e1f79a70da55554a316af